lgc/builder/BuilderReplayer.cpp

```diff
@@ -197,15 +197,15 @@ void BuilderReplayer::replayCall(unsigned opcode, CallInst *call) {
   // Process the builder call.
   LLVM_DEBUG(dbgs() << "Replaying " << *call << "\n");
   Value *newValue = processCall(opcode, call);
+  if (!newValue)
+    newValue = UndefValue::get(call->getType());
```

Comment:
Why is that necessary?

Reply:
This function is called in below code
```
 while (!func.use_empty()) {
      CallInst *call = cast<CallInst>(func.use_begin()->getUser());
      replayCall(opcode, call);
    }
```
So we must erase the `call` instruction, otherwise we will entry a endless. but we must replace this `call` before erasing, so undefined value is necessary.

Reply:
Hmm, does this mean that there is some other code that expects the replay to be successful and checks explicitly for an `undef` value? That's unfortunate and should really be fixed, because the practice of assigning specific semantic meaning to `undef` doesn't mesh well with how LLVM IR defined and should be avoided, and we should not expand the behavior of relying on `undef` in this way.

Is it easy to fix the larger issue here? If yes, that would be great, but if it isn't then feel free to go back to the previous version where `undef` was created by `findResourceNode`.

Do you have e.g. a CTS test that is affected by this problem and needs the `undef`?

Reply:
There is no code to check `undef`, we just need to replace the function with return value here, but `nullptr` can't be as a new value.

I will try to fix this issue next week. and revert to the previous version first.

You can launch the CTS test 'dEQP-VK.api.copy_and_blit.copy_commands2.resolve_image.whole_array_image.4_bit',  the issue will be observed.

Thanks.